src/core/hle/libraries/libkernel/libkernel.cpp

```diff
@@ -28,11 +32,79 @@ int32_t PS4_SYSV_ABI sceKernelReleaseDirectMemory(off_t start, size_t len) {
 
 static PS4_SYSV_ABI void stack_chk_fail() { BREAKPOINT(); }
 
-int PS4_SYSV_ABI sceKernelMunmap(void* addr, size_t len) { BREAKPOINT(); }
+int PS4_SYSV_ABI sceKernelMunmap(void* addr, size_t len) {
+    BREAKPOINT();
+    return SCE_OK;
+}
+
+void PS4_SYSV_ABI sceKernelUsleep(unsigned int microseconds) {
+    std::this_thread::sleep_for(std::chrono::microseconds(microseconds));
+}
+
+struct iovec {
+    void* iov_base; /* Base	address. */
+    size_t iov_len; /* Length. */
+};
+
+size_t PS4_SYSV_ABI _writev(int fd, const struct iovec* iov, int iovcn) {
+    //weird it gives fd ==0 and writes to stdout , i am not sure if it that is valid (found in openorbis)
+    size_t total_written = 0;
+    for (int i = 0; i < iovcn; i++) {
+        total_written += ::fwrite(iov[i].iov_base, 1, iov[i].iov_len, stdout);
+    }
+    return total_written;
+}
 
 static thread_local int libc_error;
 int* PS4_SYSV_ABI __Error() { return &libc_error; }
 
+#define PROT_READ 0x1
+#define PROT_WRITE 0x2
+
+int PS4_SYSV_ABI sceKernelMmap(void* addr, u64 len, int prot, int flags, int fd, off_t offset, void** res) {
+    PRINT_FUNCTION_NAME();
+    if (prot > 3)// READ,WRITE or bitwise READ | WRITE supported
+    {
+        LOG_ERROR_IF(log_libkernel_file, "sceKernelMmap prot ={} not supported\n",prot);
+    }
+    DWORD flProtect;
```

Comment:
If PROT_WRITE is not specified, this will have a garbage value. Initialize it appropriately

Reply:
hmm?